test/evergreen.yml

```diff
@@ -565,7 +565,41 @@ functions:
         pip3 install lxml==4.8.0 Pygments==2.11.2 Jinja2==3.0.3 gcovr==5.0
         mkdir -p ../coverage_report
         GCOV=/opt/mongodbtoolchain/v4/bin/gcov gcovr -r .. -f ../src -e '.*/bt_(debug|dump|misc|salvage|vrfy).*' -e '.*/(log|progress|verify_build|strerror|env_msg|err_file|cur_config|os_abort)\..*' -e '.*_stat\..*' -e 'bench' -e 'examples' -e 'test' -e 'ext' -e 'dist' -e 'tools' -j 4 --html-self-contained --html-details ../coverage_report/2_coverage_report.html --json-summary-pretty --json-summary ../coverage_report/1_coverage_report_summary.json
-        python3 ../test/evergreen/code_coverage_analysis.py -c ../coverage_report/1_coverage_report_summary.json -t ../time.txt
+        python3 ../test/evergreen/code_coverage_analysis.py -s ../coverage_report/1_coverage_report_summary.json -t ../time.txt
+
+  "component code coverage analysis":
+    command: shell.exec
+    params:
+      working_dir: "wiredtiger/cmake_build"
+      shell: bash
+      script: |
+        set -o errexit
+        set -o verbose
+        virtualenv -p python3 venv
+        source venv/bin/activate
+        pip3 install lxml==4.8.0 Pygments==2.11.2 Jinja2==3.0.3 gcovr==5.0
+        mkdir -p ../coverage_report
+        python3 ../test/evergreen/code_coverage_analysis.py -c component_coverage -s ../coverage_report/1_coverage_report_summary.json -t ../time.txt -o ../atlas_out_code_coverage.json
+
+  "upload-component-coverage-report-atlas":
```

Comment:
I believe the function naming convention is to avoid using "-".

Reply:
Looks like most of the contents of this new function are the same as the existing "upload-perf-test-stats" function. Can we reuse the existing function?

Reply:
I updated to use the existing function "upload-perf-test-stats" but could not find a way to skip the `perf.send` command because in the code coverage, we do not generate evergreen_out_XXX.json file.

Reply:
Can't you use a function argument to distinguish between perf and coverage tests, and only call `perf.send` for the former test?

Reply:
I'd love to avoid code duplication by abstracting the command code segment into a function. If it turns out non-trivial to achieve that or makes the existing function harder to read, I'm okay to leave it as is. 

I do think the function deserves a better name - it's not actually confined to uploading "component-only" coverage report. I'd expect the same function can be reused for the upcoming complexity metric as well.

Reply:
Fixed in c28fff8cf1be812fc473f63c57414d9e14f82e66. Broke down the function `upload-perf-test-stats` to "upload-stats-to-atlas", "upload-stats-to-evergreen" and used "upload-stats-to-atlas" in coverage report to upload the stats.

Reply:
I like this change!

> I believe the function naming convention is to avoid using "-".

I think this comment raised before still applies.

Reply:
Thanks, Fixed in 434e7a02f3463a481628283157c70772311c1733